[master] Add syncing of custom ssh wrappers

lkubb opened the pull request:
### What does this PR do?
Adds `sync_wrapper` to the `saltutil` runner/execution/state modules.
From what I can tell, the loader is already configured to load them.

### What issues does this PR fix or reference?
Fixes: https://github.com/saltstack/salt/issues/45450
Reference: https://github.com/saltstack/salt/issues/61637

### Previous Behavior
`<cache_dir>/extmods/wrapper` is respected by the loader, but cannot be synced from the fileserver

### New Behavior
`<cache_dir>/extmods/wrapper` can be synced

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**
<!-- Please review the [test documentation](https://docs.saltproject.io/en/master/topics/tutorials/writing_tests.html) for details on how to implement tests into Salt's test suite. -->
- [x] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

lkubb committed:
Add syncing of custom ssh wrappers

lkubb committed:
run pre-commit upgrade code for Py3.8+

lkubb committed:
Add `sync_wrapper` to `saltutil` state module

lkubb committed:
run pre-commit upgrade code for Py3.8+ again

Ch3LL commented on salt/runners/saltutil.py:
```diff
@@ -835,3 +840,34 @@ def sync_executors(saltenv="base", extmod_whitelist=None, extmod_blacklist=None)
+
+def sync_wrapper(saltenv="base", extmod_whitelist=None, extmod_blacklist=None):
+    """
+    .. versionadded:: 3000
```
```suggestion
    .. versionadded:: 3007.0
```

Ch3LL commented on salt/states/saltutil.py:
```diff
@@ -349,3 +349,17 @@ def sync_serializers(name, **kwargs):
+
+def sync_wrapper(name, **kwargs):
+    """
+    Performs the same task as saltutil.sync_wrapper module
```
```suggestion
    .. versionadded:: 3007.0
    
    Performs the same task as saltutil.sync_wrapper module
```

Ch3LL reviewed (changes requested):
Could you add a test that syncs a custom salt-ssh wraper and then salt-ssh actually uses that wrapper?

Also, this is great, thanks once again.

lkubb committed:
Correct/add versionadded

lkubb committed:
Add test for custom wrapper usage

lkubb commented:
Whoops, I was a bit too hasty in creating this PR, sorry. It was probably the surprise to find that 95% of the functionality was there already...

`versionadded` is now corrected and I added a test for syncing + usage. It works as expected on my machine, yay.
